### DISCUSSION TITLE
Every Request From Localhost?

### DISCUSSION THREAD

**Discussion by darrylmorley** [caprover/caprover#1973] [Q&A] [answered]
Hi,

I've just set up a basic Express API, I'm using Morgan for logs and trying to use a CORS whitelist. The problem is that every request comes from localhost?

e.g.:

`2024-01-02T18:47:10.814991752Z ::ffff:10.0.1.11 - - [02/Jan/2024:18:47:10 +0000] "GET / HTTP/1.0" 200 4374043 "-" "insomnia/7.1.1"`

How can I ensure that the client IP is passed and picked up by the cors middleware?

**Comment by darrylmorley** [comment]
I've implemented a workaround for this so far, but is it possible to receive the origin header?

**Comment by thgh** [comment, marked as answer]
You may be looking for `app.set('trust proxy', true)`

**Reply by darrylmorley** [reply]
Thanks, I think this is exactly what I'm looking for!
